### PR TITLE
prov/rxm: refactor and fix issues on error path

### PR DESCRIPTION
**Pull request by a-ilango**

### CODE REVIEW

File: prov/rxm/src/rxm_ep.c [hunk 1]
```diff
@@ -964,14 +963,20 @@ void rxm_ep_handle_postponed_tx_op(struct rxm_ep *rxm_ep,
 					 rxm_pkt_size + sizeof(*rma_iov) +
 					 sizeof(*rma_iov->iov) * tx_entry->count);
 		if (OFI_UNLIKELY(ret)) {
-			comp.op_context = tx_entry->context;
-			rxm_cq_write_error(rxm_ep->msg_cq, &comp, ret);
+			FI_WARN(&rxm_prov, FI_LOG_EP_DATA,
+				"Unable to perform deffered large send operation\n");
```
Comment (shefty) [inline prov/rxm/src/rxm_ep.c]: nit: deferred spelling -- also below

File: prov/rxm/src/rxm_cq.c [hunk 2]
```diff
@@ -594,8 +594,22 @@ static ssize_t rxm_cq_handle_comp(struct rxm_ep *rxm_ep,
...
+			void *op_context, int err)
+{
+	struct fi_cq_err_entry err_entry = {0};
+	err_entry.op_context = op_context;
+	err_entry.prov_errno = err;
+	err_entry.err = err;
+
+	rxm_cntr_incerr(cntr);
+	if (ofi_cq_write_error(cq, &err_entry))
+		assert(0);
```
Comment (shefty) [inline prov/rxm/src/rxm_cq.c]: I would add a log message.

File: prov/rxm/src/rxm_cq.c [hunk 3]
```diff
@@ -607,39 +607,53 @@ void rxm_cq_write_error(struct util_cq *cq, struct util_cntr *cntr,
...
+	err_entry.err = err;
+	if (rxm_ep->util_ep.tx_cq) {
+		ret = ofi_cq_write_error(rxm_ep->util_ep.tx_cq, &err_entry);
+		if (ret)
+			assert(0);
+	}
+	if (rxm_ep->util_ep.rx_cq) {
+		ret = ofi_cq_write_error(rxm_ep->util_ep.rx_cq, &err_entry);
+		if (ret)
+			assert(0);
```
Comment (shefty) [inline prov/rxm/src/rxm_cq.c]: add log messages on cq write error failures.  The commit message should be updated to reflect what's actually happening -- we're getting an error trying to read a cq error and have no idea what else to do.

File: prov/rxm/src/rxm_cq.c [hunk 4]
```diff
@@ -607,39 +607,53 @@ void rxm_cq_write_error(struct util_cq *cq, struct util_cntr *cntr,
...
+	}
+	if (rxm_ep->util_ep.flags & OFI_CNTR_ENABLED) {
+		rxm_cntr_incerr(rxm_ep->util_ep.tx_cntr);
+		rxm_cntr_incerr(rxm_ep->util_ep.rx_cntr);
+		rxm_cntr_incerr(rxm_ep->util_ep.wr_cntr);
+		rxm_cntr_incerr(rxm_ep->util_ep.rd_cntr);
```
Comment (shefty) [inline prov/rxm/src/rxm_cq.c]: This isn't necessary true, though.  Not all counters should increment.

Comment (a-ilango) [reply prov/rxm/src/rxm_cq.c]: Yes but we don't have the context to determine which counter to increment. Also, I should check if a counter is valid before incrementing it. I'll add it.

File: prov/rxm/src/rxm_cq.c (hunk 3 above)
Comment (a-ilango) [reply prov/rxm/src/rxm_cq.c]: 👍

File: prov/rxm/src/rxm_cq.c (hunk 4 above)
Comment (shefty) [reply prov/rxm/src/rxm_cq.c]: Yeah, I figured out what was happening after writing the comment.

**Pull request merged**